src/handle.jl

```diff
@@ -29,25 +29,33 @@ typealias KINMemPtr Ptr{KINMem}
    Manages automatic destruction of the referenced objects when it is
    no longer in use.
 """
-immutable Handle{T <: AbstractSundialsObject}
+immutable Handle{T}
     ptr_ref::Ref{Ptr{T}} # pointer to a pointer
 
-    @compat function (::Type{Handle}){T}(ptr::Ptr{T})
+    @compat function (::Type{Handle}){T<: AbstractSundialsObject}(ptr::Ptr{T})
         (ptr == C_NULL) && throw(ArgumentError("Null pointer passed to Handle()"))
         h = new{T}(Ref{Ptr{T}}(ptr))
         finalizer(h.ptr_ref, release_handle)
         return h
     end
+    @compat function (::Type{Handle})(ptr::Ptr{Void})
```

Comment:
It's not necessary, because if e.g. `CVodeCreate()` returns null, it would be of type `CVODEMemPtr===Ptr{CVODEMem}`. Allowing `Void` or any other non `AbstractSundialsObject` types would just hide potential misuses of the `Handle`. Also, it makes Sundials type-unstable with a big impact on performance.